src/Microsoft.Azure.AppConfiguration.AspNetCore/AzureAppConfigurationRefreshMiddleware.cs

```diff
@@ -24,9 +25,15 @@ public AzureAppConfigurationRefreshMiddleware(RequestDelegate next, IConfigurati
 
         public async Task InvokeAsync(HttpContext context)
         {
-            foreach (var refresher in Refreshers)
+            //
+            // Configuration refresh is meant to execute as an isolated background task.
+            // To prevent access of request-based resources, such as HttpContext, we suppress the execution context within the refresh operation.
+            using (AsyncFlowControl flowControl = ExecutionContext.SuppressFlow())
             {
-                _ = refresher.TryRefreshAsync();
+                foreach (IConfigurationRefresher refresher in Refreshers)
+                {
+                    _ = Task.Run(() => refresher.TryRefreshAsync());
+                }
             }
```

Comment:
This seems like an unnecessary amount of work to do on every request. Assume the server is running at 100k RPS. I assume the refreshers themselves are throttled? You still want a check in the middleware to call them at most every 60s(?).

Reply:
It is mostly no-op. Refresh only takes action on user-configured expiration interval (default 30s).